index.js

```diff
@@ -215,64 +217,65 @@ AFRAME.registerComponent('teleport-controls', {
-    newRigWorldPosition.copy(this.hitPoint);
+      const rig = this.data.cameraRig || this.el.sceneEl.camera.el;
+      rig.object3D.getWorldPosition(rigWorldPosition);
+      newRigWorldPosition.copy(this.hitPoint);
 
-    // If a teleportOrigin exists, offset the rig such that the teleportOrigin is above the hitPoint
-    const teleportOrigin = this.data.teleportOrigin;
-    if (teleportOrigin) {
-      teleportOrigin.object3D.getWorldPosition(teleportOriginWorldPosition);
-      newRigWorldPosition.sub(teleportOriginWorldPosition).add(rigWorldPosition);
-    }
+      // If a teleportOrigin exists, offset the rig such that the teleportOrigin is above the hitPoint
+      const teleportOrigin = this.data.teleportOrigin;
+      if (teleportOrigin) {
+        teleportOrigin.object3D.getWorldPosition(teleportOriginWorldPosition);
+        newRigWorldPosition.sub(teleportOriginWorldPosition).add(rigWorldPosition);
+      }
 
-    // Always keep the rig at the same offset off the ground after teleporting
-    newRigWorldPosition.y = rigWorldPosition.y + this.hitPoint.y - this.prevHitHeight;
-    this.prevHitHeight = this.hitPoint.y;
+      // Always keep the rig at the same offset off the ground after teleporting
+      newRigWorldPosition.y = rigWorldPosition.y + this.hitPoint.y - this.prevHitHeight;
+      this.prevHitHeight = this.hitPoint.y;
 
-    // Finally update the rigs position
-    newRigLocalPosition.copy(newRigWorldPosition);
-    if(rig.object3D.parent) {
-      rig.object3D.parent.worldToLocal(newRigLocalPosition);
-    }
-    rig.setAttribute('position', newRigLocalPosition);
-
-    // If a rig was not explicitly declared, look for hands and mvoe them proportionally as well
-    if (!this.data.cameraRig) { 
-      var hands = document.querySelectorAll('a-entity[tracked-controls]');
-      for (var i = 0; i < hands.length; i++) {
-        var position = hands[i].getAttribute('position');
-        var diff =rigWorldPosition.clone().sub(position);
-        var newPosition = newRigWorldPosition.clone().sub(diff);
-        hands[i].setAttribute('position', newPosition);
+      // Finally update the rigs position
+      newRigLocalPosition.copy(newRigWorldPosition);
+      if (rig.object3D.parent) {
+        rig.object3D.parent.worldToLocal(newRigLocalPosition);
+      }
+      rig.setAttribute('position', newRigLocalPosition);
+
+      // If a rig was not explicitly declared, look for hands and mvoe them proportionally as well
+      if (!this.data.cameraRig) {
+        var hands = document.querySelectorAll('a-entity[tracked-controls]');
+        for (var i = 0; i < hands.length; i++) {
+          var position = hands[i].getAttribute('position');
+          var diff = auxVector3.copy(rigWorldPosition).sub(position);
+          var newPosition = auxVector3.copy(newRigWorldPosition).sub(diff);
+          hands[i].setAttribute('position', newPosition);
+        }
       }
-    }
 
-    this.el.emit('teleport', {
```

Comment:
Could save another object by storing `this.teleportEventDetail` and updating that. If you keep `this.rigWorldPosition`, `this.rigWorldPosition`, `this.hitPoint`, you don' even have to update it since it's referenced.

Reply:
Yep good idea :+1:

Reply:
Ops I just messed up without pushing these changes, I'll open a new one :)